### PR TITLE
factory contract for ERC20&IRC2

### PR DESCRIPTION
**Pull request by quocle108**

### CODE REVIEW

File: doc/deployment.md [hunk 1]
```diff
@@ -1070,24 +1070,48 @@ goloop rpc --uri https://btp.net.solidwallet.io/api/v3/ sendtx call --to cx11a5a
...
+    --param _irc31=cxc1e92e175e1e5f98edf62b192ae051caae994a97 \
+    --param _name=ICX \
+    --param _serializedIrc2=$IRC2_SERIALIZED
 
-goloop rpc --uri https://btp.net.solidwallet.io/api/v3/ txresult 0xbfb1c886274b972534d9cccd3e5b07f5ab6258428400d5de4e5b39232fa00236
+goloop rpc --uri https://btp.net.solidwallet.io/api/v3/ txresult 0xdd53327a0f5b5e2b433c49ec43d8c9f45b54295de81b3cd74db99be75257810c
 ```
 
-#### Deploy BSH
+#### Deploy IRC2 token Jar
```
Comment (simsonraj) [inline doc/deployment.md]: I suppose we can remove these sections now anything related to IRC2 token Jar

File: javascore/irc2Tradeable/src/main/java/foundation/icon/btp/irc2Tradeable/IRC2Basic.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        Context.require(safeGetBalance(_from).compareTo(_value) >= 0, "Insufficient balance");
+
+        // adjust the balances
+        safeSetBalance(_from, safeGetBalance(_from).subtract(_value));
+        safeSetBalance(_to, safeGetBalance(_to).add(_value));
+
+        // if the recipient is SCORE, call 'tokenFallback' to handle further operation
+        byte[] dataBytes = (_data == null) ? new byte[0] : _data;
+        if (_to.isContract()) {
+            Context.call(_to, "tokenFallback", _from, _value, dataBytes);
```
Comment (simsonraj) [inline javascore/irc2Tradeable/src/main/java/foundation/icon/btp/irc2Tradeable/IRC2Basic.java:102]: Just making sure, the BSH contract still needs the tokenFallBack to maintain the balances, and the BSH has not been changed to get the balances from the IRC2 contract, if I'm not wrong?

File: javascore/irc2Tradeable/src/main/java/foundation/icon/btp/irc2Tradeable/IRC2Tradeable.java [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    private final Address ZERO_ADDRESS = Address.fromString("hx0000000000000000000000000000000000000000");
+
+    private final OwnerManager ownerManager = new OwnerManagerImpl("owners");
+
+    public IRC2Tradeable(String _name, String _symbol, int _decimals) {
+        super(_name, _symbol, _decimals);
+    }
+
+    @External
+    public void burn(BigInteger _amount) {
```
Comment (simsonraj) [inline javascore/irc2Tradeable/src/main/java/foundation/icon/btp/irc2Tradeable/IRC2Tradeable.java:41]: Does the burn needs the onlyOwner, similar to the Solidity?

**Review by simsonraj — COMMENTED** [review]
I haven't set up in the local, but the changes look good to me on glancing at it. There are still instances in the contracts which say ERC1155/IRC31, if we can change them to ERC20/IRC20 will be perfect.

File: javascore/irc2Tradeable/src/main/java/foundation/icon/btp/irc2Tradeable/IRC2Tradeable.java (hunk 3 above)
Comment (quocle108) [reply javascore/irc2Tradeable/src/main/java/foundation/icon/btp/irc2Tradeable/IRC2Tradeable.java:41]: You're right. This token should only burn by the Owner. Normal users should not allow burning their token

File: doc/deployment.md (hunk 1 above)
Comment (quocle108) [reply doc/deployment.md]: yeah, i am going to remote it

File: javascore/irc2Tradeable/src/main/java/foundation/icon/btp/irc2Tradeable/IRC2Basic.java (hunk 2 above)
Comment (quocle108) [reply javascore/irc2Tradeable/src/main/java/foundation/icon/btp/irc2Tradeable/IRC2Basic.java:102]: not sure i can understand

File: javascore/irc2Tradeable/src/main/java/foundation/icon/btp/irc2Tradeable/IRC2Tradeable.java (hunk 3 above)
Comment (simsonraj) [reply javascore/irc2Tradeable/src/main/java/foundation/icon/btp/irc2Tradeable/IRC2Tradeable.java:41]: This is all I have, Rest of the things looking good to me, Once you add the OnlyOwner, ill resolve this

**Pull request merged**